{
  "url": "https://api.github.com/repos/shap/shap/issues/1392",
  "repository_url": "https://api.github.com/repos/shap/shap",
  "labels_url": "https://api.github.com/repos/shap/shap/issues/1392/labels{/name}",
  "comments_url": "https://api.github.com/repos/shap/shap/issues/1392/comments",
  "events_url": "https://api.github.com/repos/shap/shap/issues/1392/events",
  "html_url": "https://github.com/shap/shap/issues/1392",
  "id": 688598193,
  "node_id": "MDU6SXNzdWU2ODg1OTgxOTM=",
  "number": 1392,
  "title": "Import error on modules from shap.plots",
  "user": {
    "login": "rsarai",
    "id": 17428455,
    "node_id": "MDQ6VXNlcjE3NDI4NDU1",
    "avatar_url": "https://avatars.githubusercontent.com/u/17428455?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/rsarai",
    "html_url": "https://github.com/rsarai",
    "followers_url": "https://api.github.com/users/rsarai/followers",
    "following_url": "https://api.github.com/users/rsarai/following{/other_user}",
    "gists_url": "https://api.github.com/users/rsarai/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/rsarai/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/rsarai/subscriptions",
    "organizations_url": "https://api.github.com/users/rsarai/orgs",
    "repos_url": "https://api.github.com/users/rsarai/repos",
    "events_url": "https://api.github.com/users/rsarai/events{/privacy}",
    "received_events_url": "https://api.github.com/users/rsarai/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [

  ],
  "state": "closed",
  "locked": false,
  "assignee": null,
  "assignees": [

  ],
  "milestone": null,
  "comments": 3,
  "created_at": "2020-08-29T20:19:19Z",
  "updated_at": "2020-09-03T11:20:31Z",
  "closed_at": "2020-09-03T11:20:31Z",
  "author_association": "NONE",
  "active_lock_reason": null,
  "body": "Hello folks, \r\n\r\nI'm trying to use the `heatmap` plot and I'm having a few import issues:\r\n```python\r\nimport shap\r\nshap.plots.heatmap(shap_values)\r\n# AttributeError: module 'shap.plots' has no attribute 'heatmap'\r\n```\r\n\r\nI also tried to import the module directly:\r\n```python\r\nimport shap.plots._heatmap\r\nheatmap(shap_values)\r\n\r\n# ----> 8 from .. import Explanation\r\n# ImportError: cannot import name 'Explanation' from 'shap' \r\n```\r\n\r\nAm I missing something? Is there any reason specific reason to use relative imports [here](https://github.com/slundberg/shap/blob/df4334a700e419a0623acde497c12e48029946fe/shap/plots/_heatmap.py#L8). From what I know these import problems can be mitigated by replacing relative imports with absolute imports\r\n\r\nI'm using the version: 0.36.0",
  "closed_by": {
    "login": "rsarai",
    "id": 17428455,
    "node_id": "MDQ6VXNlcjE3NDI4NDU1",
    "avatar_url": "https://avatars.githubusercontent.com/u/17428455?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/rsarai",
    "html_url": "https://github.com/rsarai",
    "followers_url": "https://api.github.com/users/rsarai/followers",
    "following_url": "https://api.github.com/users/rsarai/following{/other_user}",
    "gists_url": "https://api.github.com/users/rsarai/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/rsarai/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/rsarai/subscriptions",
    "organizations_url": "https://api.github.com/users/rsarai/orgs",
    "repos_url": "https://api.github.com/users/rsarai/repos",
    "events_url": "https://api.github.com/users/rsarai/events{/privacy}",
    "received_events_url": "https://api.github.com/users/rsarai/received_events",
    "type": "User",
    "site_admin": false
  },
  "reactions": {
    "url": "https://api.github.com/repos/shap/shap/issues/1392/reactions",
    "total_count": 0,
    "+1": 0,
    "-1": 0,
    "laugh": 0,
    "hooray": 0,
    "confused": 0,
    "heart": 0,
    "rocket": 0,
    "eyes": 0
  },
  "timeline_url": "https://api.github.com/repos/shap/shap/issues/1392/timeline",
  "performed_via_github_app": null,
  "state_reason": "completed"
}
